src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerFileReference.cs

```diff
@@ -118,38 +113,27 @@ public override string FullPath
             }
         }
 
-        private static string GetAssemblyNameFromPath(string path)
+        public override string Display
         {
-            // AssemblyName.GetAssemblyName(path) is not available on CoreCLR.
-            // Use our metadata reader to do the equivalent thing.
-            using (var reader = new PEReader(FileUtilities.OpenRead(path)))
+            get
             {
-                var metadataReader = reader.GetMetadataReader();
-                var assemblyDefinition = metadataReader.GetAssemblyDefinition();
-                return metadataReader.GetString(assemblyDefinition.Name);
+                return _displayName;
             }
         }
 
-        public override string Display
+        public override string ID
         {
             get
             {
-                if (_lazyDisplayName == null)
-                {
-                    try
-                    {
-                        _lazyDisplayName = GetAssemblyNameFromPath(_fullPath);
-                    }
-                    catch (Exception)
-                    { }
-
-                    _lazyDisplayName = _lazyDisplayName ?? Path.GetFileName(this.FullPath);
-                }
-
-                return _lazyDisplayName;
+                return _assembly.FullName;
             }
         }
 
+        public Assembly GetAssembly()
```

Comment:
Is there a reason this is a method and not a property?  

Is this just historical because we used to delay load the `Assembly`?  If so I'd consider changing this to a property.